modules/xmpp/xmpp.js

```diff
@@ -509,4 +512,45 @@ export default class XMPP extends Listenable {
         initRayo();
         initStropheLogger();
     }
+
+    /**
+     * Returns details about connection failure. Shard change or is it after
+     * suspend.
+     * @returns {object} contains details about a connection failure.
+     * @private
+     */
+    _getConnectionFailedReasonDetails() {
+        const details = {};
+
+        // check for moving between shard if information is available
+        if (this.options.deploymentInfo
+            && this.options.deploymentInfo.shard
+            && this.connection._proto
+            && this.connection._proto.lastResponseHeaders) {
+
+            // split headers by line
+            const headersArr = this.connection._proto.lastResponseHeaders
+                .trim().split(/[\r\n]+/);
+            const headers = {};
+
+            headersArr.forEach(line => {
+                const parts = line.split(': ');
+                const header = parts.shift();
+                const value = parts.join(': ');
+
+                headers[header] = value;
+            });
+
+            /* eslint-disable camelcase */
+            details.shard_changed
+                = this.options.deploymentInfo.shard
+                    !== headers['x-jitsi-shard'];
+            /* eslint-enable camelcase */
+        }
+
+        // check for possible suspend
```

Comment:
I thought we also wanted the suspend duration somewhere

Reply:
Ok, I will modify it and add that.